src/_pytest/assertion/util.py

```diff
@@ -169,14 +176,22 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
 
 
 def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+    """
+    Generic comparison function, dispatches to other comparison functions based on the inputs' type.
+    :param left: 1st object passed for comparison.
+    :param right: 2nd object passed for comparison.
+    :param verbose: decides whether to print full error report or only a part of it.
+    """
+
+    explanation = []
+
```

Comment:
This initialization is not needed:

```suggestion

    explanation = []

```